packages/flutter/lib/src/material/bottom_sheet_theme.dart

```diff
@@ -67,6 +68,11 @@ class BottomSheetThemeData with Diagnosticable {
   /// If null, [BottomSheet] uses [Clip.none].
   final Clip? clipBehavior;
 
+  /// Constrains the size of the [BottomSheet].
+  ///
+  /// If null, the bottom sheet's size will be unconstrained.
```

Comment:
... and if it is constrained the bottom sheet will be aligned with the bottom-center of the available space...